spinalcordtoolbox/download.py

```diff
@@ -367,21 +385,80 @@ def install_named_dataset(dataset_name, dest_folder=None, keep=False):
                          f"{sorted(list(DATASET_DICT.keys()), key=str.casefold)}")
 
     urls = DATASET_DICT[dataset_name]["mirrors"]
+
+    custom = True
     if dest_folder is None:
+        custom = False
         dest_folder = DATASET_DICT[dataset_name]["default_location"]
 
     install_data(urls, dest_folder, keep)
 
+    if custom is True:
+        record_custom_folders(dest_folder, dataset_name)
+
+
+def is_installed(dataset_name, custom_paths):
+    """
+    :return: bool: Checks whether a dataset is installed
+    """
+    # Check for installation at default location
+    path_default = DATASET_DICT[dataset_name]['default_location']
+    if os.path.exists(path_default):
+        return True
```

Comment:
One additional wrinkle: Our binary files get installed directly to `bin/`, rather than a subfolder (like the other `data/` downloads). This is because `bin/` gets added to the `$PATH`, and so it's easiest to keep all executable files in a single directory.

This presents a problem, though, because `bin/` is used for other files, too. (Namely, the launcher entry points for SCT's CLI scripts). Because `bin/` is a multi-purpose directory for all executable files, it's possible for `bin/` to exist, yet not contain the necessary binaries. Meaning, a simple directory check doesn't quite work.

I'm not quite sure how to handle this without diving into a more complex solution (e.g. tracking the archive contents / hashing the files), which would expand the scope of this PR and get into previous discussions such as:

- https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3380
- https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/2959
- https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3901